internal/datacoord/compaction_trigger.go

```diff
@@ -552,6 +566,80 @@ func (t *compactionTrigger) ShouldDoSingleCompaction(segment *SegmentInfo, compa
 	return false
 }
 
+func filterInIndexedSegments(meta *meta, indexCoord types.IndexCoord, segments ...*SegmentInfo) []*SegmentInfo {
+	if len(segments) == 0 {
+		return nil
+	}
+
+	segmentMap := make(map[int64]*SegmentInfo)
+	collectionSegments := make(map[int64][]*SegmentInfo)
+	vecFieldID := make(map[int64]int64)
```

Comment:
Maybe add a `TODO` note here, since this implementation cannot process multiple vector field cases.